src/modules/launcher/Plugins/Microsoft.Plugin.Uri/BrowserInfo.cs

```diff
@@ -0,0 +1,119 @@
+
+                // "Handles 'Indirect Strings' (UWP programs)"
+                // Using Ordinal since this is internal and used with a symbol
+                if (programLocation.StartsWith("@", StringComparison.Ordinal))
+                {
+                    var directProgramLocationStringBuilder = new StringBuilder(128);
+                    if (NativeMethods.SHLoadIndirectString(
+                            programLocation,
+                            directProgramLocationStringBuilder,
+                            (uint)directProgramLocationStringBuilder.Capacity,
+                            IntPtr.Zero) ==
+                        NativeMethods.Hresult.Ok)
+                    {
+                        // Check if there's a postfix with contract-white/contrast-black icon is available and use that instead
+                        var directProgramLocation = directProgramLocationStringBuilder.ToString();
+                        var themeIcon = newTheme == ManagedCommon.Theme.Light || newTheme == ManagedCommon.Theme.HighContrastWhite
+                            ? "contrast-white"
+                            : "contrast-black";
+                        var extension = FilePath.GetExtension(directProgramLocation);
+                        var themedProgLocation =
+                            $"{directProgramLocation.Substring(0, directProgramLocation.Length - extension.Length)}_{themeIcon}{extension}";
+                        IconPath = File.Exists(themedProgLocation)
+                            ? themedProgLocation
+                            : directProgramLocation;
+
+                        Path = string.Equals(FilePath.GetExtension(directProgramLocation), ".exe", StringComparison.Ordinal)
+                            ? directProgramLocation
+                            : null;
+                    }
+
+                    throw new Exception("Could not load indirect string.");
+                }
+                else
+                {
+                    // Using Ordinal since this is internal and used with a symbol
+                    var indexOfComma = programLocation.IndexOf(',', StringComparison.Ordinal);
+                    IconPath = indexOfComma > 0
+                        ? programLocation.Substring(0, indexOfComma)
+                        : programLocation;
+                    Path = IconPath;
+                }
+
+                if (string.IsNullOrEmpty(Path))
+                {
+                    throw new Exception("Browser path is null or empty.");
+                }
+            }
+            catch (Exception e)
+            {
+                if (defaultToEdgeOnFail)
+                {
+                    Path = MSEdgePath;
+                    IconPath = MSEdgePath;
+                }
+                else
+                {
+                    Path = null;
+                    IconPath = null;
```

Comment:
`=string.Empty` might be better here and probably won't throw an param null exception. I am not a friend of setting strong to null.

Reply:
And I personally don't think that we need this case at any time.

Reply:
If `Path` is null because the default browser is not set, and we open a link (in shell) it will prompt the user to select a browser. Thats why this case is the default behavior in the uri plugin.